dex/price-discovery/src/lib.rs

```diff
@@ -17,15 +16,14 @@ const INVALID_PAYMENT_ERR_MSG: &[u8] = b"Invalid payment token";
 const BELOW_MIN_PRICE_ERR_MSG: &[u8] = b"Launched token below min price";
 pub const MIN_PRICE_PRECISION: u64 = 1_000_000_000_000_000_000;
 
-pub struct RewardsPair<M: ManagedTypeApi> {
-    pub lp_tokens_amount: BigUint<M>,
-    pub extra_rewards_amount: BigUint<M>,
+pub struct Rewards<M: ManagedTypeApi> {
```

Comment:
you can use `EsdtTokenPayment` instead of this new struct

Reply:
Done.